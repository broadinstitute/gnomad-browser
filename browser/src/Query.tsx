import React, { Component } from 'react'

import Delayed from './Delayed'
import StatusMessage from './StatusMessage'

const areVariablesEqual = (variables: any, otherVariables: any) => {
  const keys = Object.keys(variables)
  const otherKeys = Object.keys(otherVariables)
  if (keys.length !== otherKeys.length) {
    return false
  }
  return keys.every((key) => variables[key] === otherVariables[key])
}

const cancelable = (promise: any) => {
  let isCanceled = false
  const wrapper = new Promise((resolve: any, reject: any) => {
    promise.then(
      (value: any) => {
        if (!isCanceled) {
          resolve(value)
        }
      },
      (error: any) => {
        if (!isCanceled) {
          reject(error)
        }
      }
    )
  })

  return {
    cancel: () => {
      isCanceled = true
    },
    promise: wrapper,
  }
}

type BaseQueryState = any

type BaseQueryProps = {
  operationName: string | null
  query: string
  url: string
  variables?: any
  children: (state: BaseQueryState) => JSX.Element
}

export class BaseQuery extends Component<BaseQueryProps, BaseQueryState> {
  static defaultProps = {
    url: '/api/',
    operationName: null,
    variables: {},
  }

  currentRequest: any

  mounted: any

  state = {
    data: null,
    error: null,
    graphQLErrors: null,
    loading: true,
  }

  componentDidMount() {
    this.mounted = true
    this.loadData()
  }

  componentDidUpdate(prevProps: BaseQueryProps) {
    const { query, variables } = this.props
    if (query !== prevProps.query || !areVariablesEqual(variables, prevProps.variables)) {
      this.loadData()
    }
  }

  componentWillUnmount() {
    this.mounted = false
  }

  loadData() {
    const { operationName, query, url, variables } = this.props

    this.setState({
      loading: true,
      error: null,
      graphQLErrors: null,
    })

    if (this.currentRequest) {
      this.currentRequest.cancel()
    }

    this.currentRequest = cancelable(
      fetch(url, {
        body: JSON.stringify({
          operationName,
          query,
          variables,
        }),
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
      }).then((response) => response.json())
    )
    this.currentRequest.promise.then(
      (response: any) => {
        if (!this.mounted) {
          return
        }
        this.setState({
          data: response.data,
          error: null,
          graphQLErrors: response.errors,
          loading: false,
        })
      },
      (error: any) => {
        if (!this.mounted) {
          return
        }
        this.setState({
          data: null,
          error,
          graphQLErrors: null,
          loading: false,
        })
      }
    )
  }

  render() {
    const { children } = this.props
    return children(this.state)
  }
}

type OwnQueryProps = {
  children: (...args: any[]) => any
  errorMessage?: string
  loadingMessage?: string
  loadingPlaceholderHeight?: number
  operationName: string | null
  query: string
  success?: (...args: any[]) => any
  url: string
  variables?: any
}

// @ts-expect-error TS(2456) FIXME: Type alias 'QueryProps' circularly references itse... Remove this comment to see the full error message
type QueryProps = OwnQueryProps & typeof Query.defaultProps

// @ts-expect-error TS(7022) FIXME: 'Query' implicitly has type 'any' because it does ... Remove this comment to see the full error message
const Query = ({
  children,
  errorMessage,
  loadingMessage,
  loadingPlaceholderHeight,
  operationName,
  query,
  success,
  url,
  variables,
}: QueryProps) => {
  return (
    <BaseQuery operationName={operationName} query={query} url={url} variables={variables}>
      {({ data, error, graphQLErrors, loading }: any) => {
        if (loading) {
          return (
            <div style={{ height: loadingPlaceholderHeight || 'auto' }}>
              <Delayed>
                <StatusMessage>{loadingMessage}</StatusMessage>
              </Delayed>
            </div>
          )
        }

        if (error) {
          return <StatusMessage>{errorMessage}</StatusMessage>
        }

        if (!data || !success(data)) {
          return (
            <StatusMessage>
              {graphQLErrors && graphQLErrors.length
                ? Array.from(new Set(graphQLErrors.map((e: any) => e.message))).join(', ')
                : errorMessage}
            </StatusMessage>
          )
        }

        return children({ data })
      }}
    </BaseQuery>
  )
}

Query.defaultProps = {
  errorMessage: 'Error',
  loadingMessage: 'Loading',
  loadingPlaceholderHeight: undefined,
  success: () => true,
  url: '/api/',
  variables: {},
  operationName: null,
}

export default Query
