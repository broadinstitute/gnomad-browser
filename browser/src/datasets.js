export const referenceGenomeForDataset = datasetId => {
  if (datasetId.startsWith('gnomad_r3')) {
    return 'GRCh38'
  }

  return 'GRCh37'
}

export const datasetLabels = {
  exac: 'ExAC v1.0',
  gnomad_r2_1: 'gnomAD v2.1.1',
  gnomad_r2_1_controls: 'gnomAD v2.1.1 (controls)',
  gnomad_r2_1_non_cancer: 'gnomAD v2.1.1 (non-cancer)',
  gnomad_r2_1_non_neuro: 'gnomAD v2.1.1 (non-neuro)',
  gnomad_r2_1_non_topmed: 'gnomAD v2.1.1 (non-TOPMed)',
  gnomad_r3: 'gnomAD v3.1.2',
  gnomad_r3_controls_and_biobanks: 'gnomAD v3.1.2 (controls/biobanks)',
  gnomad_r3_non_cancer: 'gnomAD v3.1.2 (non-cancer)',
  gnomad_r3_non_neuro: 'gnomAD v3.1.2 (non-neuro)',
  gnomad_r3_non_topmed: 'gnomAD v3.1.2 (non-TOPMed)',
  gnomad_r3_non_v2: 'gnomAD v3.1.2 (non-v2)',
  gnomad_sv_r2_1: 'gnomAD SVs v2.1',
  gnomad_sv_r2_1_controls: 'gnomAD SVs v2.1 (controls)',
  gnomad_sv_r2_1_non_neuro: 'gnomAD SVs v2.1 (non-neuro)',
}

export const allDatasetIds = Object.keys(datasetLabels)

export const labelForDataset = datasetId => datasetLabels[datasetId] || 'Unknown'
