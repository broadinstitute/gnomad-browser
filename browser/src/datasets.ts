import { datasetLabels } from '../../dataset-metadata/metadata'

export const referenceGenomeForDataset = (datasetId: any) => {
  if (datasetId.startsWith('gnomad_r3')) {
    return 'GRCh38'
  }

  return 'GRCh37'
}

// @ts-expect-error TS(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
export const labelForDataset = (datasetId: any) => datasetLabels[datasetId] || 'Unknown'
