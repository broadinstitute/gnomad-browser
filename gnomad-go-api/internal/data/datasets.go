package data

type DatasetConfig struct {
	ID              string
	ReferenceGenome string
	HasExome        bool
	HasGenome       bool
	HasJoint        bool
}

var Datasets = map[string]DatasetConfig{
	"gnomad_r4": {
		ID:              "gnomad_r4",
		ReferenceGenome: "GRCh38",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        true,
	},
	"gnomad_r4_non_ukb": {
		ID:              "gnomad_r4_non_ukb",
		ReferenceGenome: "GRCh38",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        true,
	},
	"gnomad_r3": {
		ID:              "gnomad_r3",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r3_controls_and_biobanks": {
		ID:              "gnomad_r3_controls_and_biobanks",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r3_non_cancer": {
		ID:              "gnomad_r3_non_cancer",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r3_non_neuro": {
		ID:              "gnomad_r3_non_neuro",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r3_non_topmed": {
		ID:              "gnomad_r3_non_topmed",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r3_non_v2": {
		ID:              "gnomad_r3_non_v2",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r2_1": {
		ID:              "gnomad_r2_1",
		ReferenceGenome: "GRCh37",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r2_1_controls": {
		ID:              "gnomad_r2_1_controls",
		ReferenceGenome: "GRCh37",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r2_1_non_cancer": {
		ID:              "gnomad_r2_1_non_cancer",
		ReferenceGenome: "GRCh37",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r2_1_non_neuro": {
		ID:              "gnomad_r2_1_non_neuro",
		ReferenceGenome: "GRCh37",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r2_1_non_topmed": {
		ID:              "gnomad_r2_1_non_topmed",
		ReferenceGenome: "GRCh37",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        false,
	},
	"exac": {
		ID:              "exac",
		ReferenceGenome: "GRCh37",
		HasExome:        true,
		HasGenome:       false,
		HasJoint:        false,
	},
	"gnomad_sv_r2_1": {
		ID:              "gnomad_sv_r2_1",
		ReferenceGenome: "GRCh37",
		HasExome:        false,
		HasGenome:       false,
		HasJoint:        false,
	},
	"gnomad_sv_r2_1_controls": {
		ID:              "gnomad_sv_r2_1_controls",
		ReferenceGenome: "GRCh37",
		HasExome:        false,
		HasGenome:       false,
		HasJoint:        false,
	},
	"gnomad_sv_r2_1_non_neuro": {
		ID:              "gnomad_sv_r2_1_non_neuro",
		ReferenceGenome: "GRCh37",
		HasExome:        false,
		HasGenome:       false,
		HasJoint:        false,
	},
	"gnomad_sv_r4": {
		ID:              "gnomad_sv_r4",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       false,
		HasJoint:        false,
	},
	"gnomad_cnv_r4": {
		ID:              "gnomad_cnv_r4",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       false,
		HasJoint:        false,
	},
}
