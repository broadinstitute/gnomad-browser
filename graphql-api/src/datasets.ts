export const DATASET_LABELS = {
  gnomad_r4: 'gnomAD v4',
  gnomad_cnv_r4: 'gnomAD v4 cnv',
  gnomad_r3: 'gnomAD v3',
  gnomad_r3_controls_and_biobanks: 'gnomAD v3 (controls and biobanks)',
  gnomad_r3_non_cancer: 'gnomAD v3 (non-cancer)',
  gnomad_r3_non_neuro: 'gnomAD v3 (non-neuro)',
  gnomad_r3_non_topmed: 'gnomAD v3 (non-TOPMed)',
  gnomad_r3_non_v2: 'gnomAD v3 (non-v2)',
  gnomad_r2_1: 'gnomAD v2',
  gnomad_r2_1_controls: 'gnomAD v2 (controls)',
  gnomad_r2_1_non_neuro: 'gnomAD v2 (non-neuro)',
  gnomad_r2_1_non_cancer: 'gnomAD v2 (non-cancer)',
  gnomad_r2_1_non_topmed: 'gnomAD v2 (non-TOPMed)',
  exac: 'ExAC',
}

export const DATASET_REFERENCE_GENOMES: Record<string, string> = {
  gnomad_r4: 'GRCh38',
  gnomad_cnv_r4: 'GRCh38',
  gnomad_r3: 'GRCh38',
  gnomad_r3_controls_and_biobanks: 'GRCh38',
  gnomad_r3_non_cancer: 'GRCh38',
  gnomad_r3_non_neuro: 'GRCh38',
  gnomad_r3_non_topmed: 'GRCh38',
  gnomad_r3_non_v2: 'GRCh38',
  gnomad_r2_1: 'GRCh37',
  gnomad_r2_1_controls: 'GRCh37',
  gnomad_r2_1_non_neuro: 'GRCh37',
  gnomad_r2_1_non_cancer: 'GRCh37',
  gnomad_r2_1_non_topmed: 'GRCh37',
  exac: 'GRCh37',
}
