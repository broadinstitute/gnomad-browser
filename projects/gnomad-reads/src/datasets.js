import logger from './logging'

const datasets = (process.env.DATASETS || '')
  .split(',')
  .filter(Boolean)
  .reduce(
    (acc, datasetId) => ({
      ...acc,
      [datasetId.toLowerCase()]: {
        exomes: {
          readsDirectory: process.env[`${datasetId.toUpperCase()}_EXOMES_READS_DIRECTORY`],
          publicPath: process.env[`${datasetId.toUpperCase()}_EXOMES_PUBLIC_PATH`],
        },
        genomes: {
          readsDirectory: process.env[`${datasetId.toUpperCase()}_GENOMES_READS_DIRECTORY`],
          publicPath: process.env[`${datasetId.toUpperCase()}_GENOMES_PUBLIC_PATH`],
        },
      },
    }),
    {}
  )

logger.info(`Dataset configuration: ${JSON.stringify(datasets)}`)

export default datasets
