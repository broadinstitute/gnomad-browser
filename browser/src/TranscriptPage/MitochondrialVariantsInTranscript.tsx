import React from 'react'

import ClinvarVariantTrack from '../ClinvarVariantsTrack/ClinvarVariantTrack'
import formatClinvarDate from '../ClinvarVariantsTrack/formatClinvarDate'
import { labelForDataset } from '../../../dataset-metadata/metadata'
import { referenceGenomeForDataset } from '../datasets'
import Link from '../Link'
import Query from '../Query'
import filterVariantsInZoomRegion from '../RegionViewer/filterVariantsInZoomRegion'
import StatusMessage from '../StatusMessage'
import { TrackPageSection } from '../TrackPage'
import MitochondrialVariants from '../MitochondrialVariantList/MitochondrialVariants'
import annotateVariantsWithClinvar from '../VariantList/annotateVariantsWithClinvar'

const query = `
query MitochondrialVariantsInTranscript($transcriptId: String!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!) {
  meta {
    clinvar_release_date
  }
  transcript(transcript_id: $transcriptId, reference_genome: $referenceGenome) {
    clinvar_variants {
      clinical_significance
      clinvar_variation_id
      gnomad {
        exome {
          ac
          an
          filters
        }
        genome {
          ac
          an
          filters
        }
      }
      gold_stars
      hgvsc
      hgvsp
      in_gnomad
      major_consequence
      pos
      review_status
      transcript_id
      variant_id
    }
    mitochondrial_variants(dataset: $datasetId) {
      ac_het
      ac_hom
      an
      consequence
      filters
      flags
      gene_id
      gene_symbol
      transcript_id
      hgvsc
      hgvsp
      lof
      lof_filter
      lof_flags
      max_heteroplasmy
      pos
      reference_genome
      variant_id
    }
  }
}
`

type OwnProps = {
  datasetId: string
  transcript: {
    transcript_id: string
  }
  zoomRegion?: {
    start: number
    stop: number
  }
}

// @ts-expect-error TS(2456) FIXME: Type alias 'Props' circularly references itself.
type Props = OwnProps & typeof MitochondrialVariantsInTranscript.defaultProps

// @ts-expect-error TS(7022) FIXME: 'MitochondrialVariantsInTranscript' implicitly has... Remove this comment to see the full error message
const MitochondrialVariantsInTranscript = ({
  datasetId,
  transcript,
  zoomRegion,
  ...rest
}: Props) => {
  if (datasetId === 'exac' || datasetId.startsWith('gnomad_r2')) {
    return (
      <StatusMessage>
        Mitochondrial variants are not available in {labelForDataset(datasetId)}
        <br />
        <br />
        <Link
          to={`/transcript/${transcript.transcript_id}?dataset=gnomad_r3`}
          preserveSelectedDataset={false}
        >
          View this transcript in gnomAD v3.1 to see mitochondrial variants
        </Link>
      </StatusMessage>
    )
  }

  return (
    <Query
      query={query}
      variables={{
        datasetId,
        transcriptId: transcript.transcript_id,
        referenceGenome: referenceGenomeForDataset(datasetId),
      }}
      loadingMessage="Loading variants"
      errorMessage="Unable to load variants"
      success={(data: any) => data.transcript && data.transcript.mitochondrial_variants}
    >
      {({ data }: any) => {
        data.transcript.mitochondrial_variants.forEach((v: any) => {
          /* eslint-disable no-param-reassign */
          if (v.an !== 0) {
            v.af = (v.ac_het + v.ac_hom) / v.an
            v.af_het = v.ac_het / v.an
            v.af_hom = v.ac_hom / v.an
          } else {
            v.af = 0
            v.af_het = 0
            v.af_hom = 0
          }
          v.hgvs = v.hgvsp || v.hgvsc
          /* eslint-enable no-param-reassign */
        })

        return (
          <>
            <TrackPageSection>
              <h2>ClinVar variants</h2>
            </TrackPageSection>
            {data.transcript.clinvar_variants.length > 0 ? (
              <>
                <ClinvarVariantTrack
                  referenceGenome={referenceGenomeForDataset(datasetId)}
                  transcripts={[transcript]}
                  variants={filterVariantsInZoomRegion(
                    data.transcript.clinvar_variants,
                    zoomRegion
                  )}
                />
                <TrackPageSection as="p">
                  Data displayed here is from ClinVar&apos;s{' '}
                  {formatClinvarDate(data.meta.clinvar_release_date)} release.
                </TrackPageSection>
              </>
            ) : (
              <TrackPageSection as="p">
                No ClinVar variants found in this transcript.
              </TrackPageSection>
            )}

            <MitochondrialVariants
              {...rest}
              clinvarReleaseDate={data.meta.clinvar_release_date}
              context={transcript}
              datasetId={datasetId}
              exportFileName={`gnomad_mitochondrial_variants_${transcript.transcript_id}`}
              variants={filterVariantsInZoomRegion(
                annotateVariantsWithClinvar(
                  data.transcript.mitochondrial_variants,
                  data.transcript.clinvar_variants
                ),
                zoomRegion
              )}
            />
          </>
        )
      }}
    </Query>
  )
}

MitochondrialVariantsInTranscript.defaultProps = {
  zoomRegion: null,
}

export default MitochondrialVariantsInTranscript
