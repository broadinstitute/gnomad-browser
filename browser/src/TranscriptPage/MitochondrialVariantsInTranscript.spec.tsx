import React from 'react'
import renderer from 'react-test-renderer'

import { jest, expect, test } from '@jest/globals'
import { mockQueries } from '../../../tests/__helpers__/queries'
import Query, { BaseQuery } from '../Query'

import MitochondrialVariantsInTranscript from './MitochondrialVariantsInTranscript'

import transcriptFactory from '../__factories__/Transcript'
import { BrowserRouter } from 'react-router-dom'
import { forAllDatasets } from '../../../tests/__helpers__/datasets'

jest.mock('../Query', () => {
  const originalModule = jest.requireActual('../Query')

  return {
    __esModule: true,
    ...(originalModule as object),
    default: jest.fn(),
    BaseQuery: jest.fn(),
  }
})

const { resetMockApiCalls, resetMockApiResponses, simulateApiResponse, setMockApiResponses } =
  mockQueries()

beforeEach(() => {
  Query.mockImplementation(
    jest.fn(({ query, children, operationName, variables }) =>
      simulateApiResponse('Query', query, children, operationName, variables)
    )
  )
  ;(BaseQuery as any).mockImplementation(
    jest.fn(({ query, children, operationName, variables }) =>
      simulateApiResponse('BaseQuery', query, children, operationName, variables)
    )
  )
})

afterEach(() => {
  resetMockApiCalls()
  resetMockApiResponses()
})

forAllDatasets('MitochondrialVariantsInTranscript with dataset %s', (datasetId) => {
  test('has no unexpected changes', () => {
    setMockApiResponses({
      MitochondrialVariantsInTranscript: () => ({
        meta: {
          clinvar_release_date: '2023-12-31',
        },
        transcript: {
          clinvar_variants: [],
          mitochondrial_variants: [],
        },
      }),
    })
    const tree = renderer.create(
      <BrowserRouter>
        <MitochondrialVariantsInTranscript
          datasetId={datasetId}
          transcript={transcriptFactory.build()}
        />
      </BrowserRouter>
    )
    expect(tree).toMatchSnapshot()
  })
})
