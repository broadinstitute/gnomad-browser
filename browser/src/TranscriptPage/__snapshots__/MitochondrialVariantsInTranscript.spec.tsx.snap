// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MitochondrialVariantsInTranscript with dataset exac has no unexpected changes 1`] = `
<div
  className="StatusMessage-sc-neg7mh-0 hJFlwG"
>
  Mitochondrial variants are not available in 
  ExAC v1.0
  <br />
  <br />
  <a
    className="Link-sc-14lgydv-0-Link jBvaYQ"
    href="/transcript/dummy_transcript?dataset=gnomad_r3"
    onClick={[Function]}
  >
    View this transcript in gnomAD v3.1 to see mitochondrial variants
  </a>
</div>
`;

exports[`MitochondrialVariantsInTranscript with dataset gnomad_cnv_r4 has no unexpected changes 1`] = `
[
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2>
      ClinVar variants
    </h2>
  </div>,
  <p
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    No ClinVar variants found in this transcript.
  </p>,
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2
      style={
        {
          "margin": "2em 0 0.25em",
        }
      }
    >
      gnomAD variants
    </h2>
    <p>
      No gnomAD variants found.
    </p>
  </div>,
]
`;

exports[`MitochondrialVariantsInTranscript with dataset gnomad_r2_1 has no unexpected changes 1`] = `
<div
  className="StatusMessage-sc-neg7mh-0 hJFlwG"
>
  Mitochondrial variants are not available in 
  gnomAD v2.1.1
  <br />
  <br />
  <a
    className="Link-sc-14lgydv-0-Link jBvaYQ"
    href="/transcript/dummy_transcript?dataset=gnomad_r3"
    onClick={[Function]}
  >
    View this transcript in gnomAD v3.1 to see mitochondrial variants
  </a>
</div>
`;

exports[`MitochondrialVariantsInTranscript with dataset gnomad_r2_1_controls has no unexpected changes 1`] = `
<div
  className="StatusMessage-sc-neg7mh-0 hJFlwG"
>
  Mitochondrial variants are not available in 
  gnomAD v2.1.1 (controls)
  <br />
  <br />
  <a
    className="Link-sc-14lgydv-0-Link jBvaYQ"
    href="/transcript/dummy_transcript?dataset=gnomad_r3"
    onClick={[Function]}
  >
    View this transcript in gnomAD v3.1 to see mitochondrial variants
  </a>
</div>
`;

exports[`MitochondrialVariantsInTranscript with dataset gnomad_r2_1_non_cancer has no unexpected changes 1`] = `
<div
  className="StatusMessage-sc-neg7mh-0 hJFlwG"
>
  Mitochondrial variants are not available in 
  gnomAD v2.1.1 (non-cancer)
  <br />
  <br />
  <a
    className="Link-sc-14lgydv-0-Link jBvaYQ"
    href="/transcript/dummy_transcript?dataset=gnomad_r3"
    onClick={[Function]}
  >
    View this transcript in gnomAD v3.1 to see mitochondrial variants
  </a>
</div>
`;

exports[`MitochondrialVariantsInTranscript with dataset gnomad_r2_1_non_neuro has no unexpected changes 1`] = `
<div
  className="StatusMessage-sc-neg7mh-0 hJFlwG"
>
  Mitochondrial variants are not available in 
  gnomAD v2.1.1 (non-neuro)
  <br />
  <br />
  <a
    className="Link-sc-14lgydv-0-Link jBvaYQ"
    href="/transcript/dummy_transcript?dataset=gnomad_r3"
    onClick={[Function]}
  >
    View this transcript in gnomAD v3.1 to see mitochondrial variants
  </a>
</div>
`;

exports[`MitochondrialVariantsInTranscript with dataset gnomad_r2_1_non_topmed has no unexpected changes 1`] = `
<div
  className="StatusMessage-sc-neg7mh-0 hJFlwG"
>
  Mitochondrial variants are not available in 
  gnomAD v2.1.1 (non-TOPMed)
  <br />
  <br />
  <a
    className="Link-sc-14lgydv-0-Link jBvaYQ"
    href="/transcript/dummy_transcript?dataset=gnomad_r3"
    onClick={[Function]}
  >
    View this transcript in gnomAD v3.1 to see mitochondrial variants
  </a>
</div>
`;

exports[`MitochondrialVariantsInTranscript with dataset gnomad_r3 has no unexpected changes 1`] = `
[
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2>
      ClinVar variants
    </h2>
  </div>,
  <p
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    No ClinVar variants found in this transcript.
  </p>,
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2
      style={
        {
          "margin": "2em 0 0.25em",
        }
      }
    >
      gnomAD variants
    </h2>
    <p>
      No gnomAD variants found.
    </p>
  </div>,
]
`;

exports[`MitochondrialVariantsInTranscript with dataset gnomad_r3_controls_and_biobanks has no unexpected changes 1`] = `
[
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2>
      ClinVar variants
    </h2>
  </div>,
  <p
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    No ClinVar variants found in this transcript.
  </p>,
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2
      style={
        {
          "margin": "2em 0 0.25em",
        }
      }
    >
      gnomAD variants
    </h2>
    <p>
      No gnomAD variants found.
    </p>
  </div>,
]
`;

exports[`MitochondrialVariantsInTranscript with dataset gnomad_r3_non_cancer has no unexpected changes 1`] = `
[
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2>
      ClinVar variants
    </h2>
  </div>,
  <p
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    No ClinVar variants found in this transcript.
  </p>,
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2
      style={
        {
          "margin": "2em 0 0.25em",
        }
      }
    >
      gnomAD variants
    </h2>
    <p>
      No gnomAD variants found.
    </p>
  </div>,
]
`;

exports[`MitochondrialVariantsInTranscript with dataset gnomad_r3_non_neuro has no unexpected changes 1`] = `
[
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2>
      ClinVar variants
    </h2>
  </div>,
  <p
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    No ClinVar variants found in this transcript.
  </p>,
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2
      style={
        {
          "margin": "2em 0 0.25em",
        }
      }
    >
      gnomAD variants
    </h2>
    <p>
      No gnomAD variants found.
    </p>
  </div>,
]
`;

exports[`MitochondrialVariantsInTranscript with dataset gnomad_r3_non_topmed has no unexpected changes 1`] = `
[
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2>
      ClinVar variants
    </h2>
  </div>,
  <p
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    No ClinVar variants found in this transcript.
  </p>,
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2
      style={
        {
          "margin": "2em 0 0.25em",
        }
      }
    >
      gnomAD variants
    </h2>
    <p>
      No gnomAD variants found.
    </p>
  </div>,
]
`;

exports[`MitochondrialVariantsInTranscript with dataset gnomad_r3_non_v2 has no unexpected changes 1`] = `
[
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2>
      ClinVar variants
    </h2>
  </div>,
  <p
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    No ClinVar variants found in this transcript.
  </p>,
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2
      style={
        {
          "margin": "2em 0 0.25em",
        }
      }
    >
      gnomAD variants
    </h2>
    <p>
      No gnomAD variants found.
    </p>
  </div>,
]
`;

exports[`MitochondrialVariantsInTranscript with dataset gnomad_r4 has no unexpected changes 1`] = `
[
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2>
      ClinVar variants
    </h2>
  </div>,
  <p
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    No ClinVar variants found in this transcript.
  </p>,
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2
      style={
        {
          "margin": "2em 0 0.25em",
        }
      }
    >
      gnomAD variants
    </h2>
    <p>
      No gnomAD variants found.
    </p>
  </div>,
]
`;

exports[`MitochondrialVariantsInTranscript with dataset gnomad_sv_r2_1 has no unexpected changes 1`] = `
[
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2>
      ClinVar variants
    </h2>
  </div>,
  <p
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    No ClinVar variants found in this transcript.
  </p>,
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2
      style={
        {
          "margin": "2em 0 0.25em",
        }
      }
    >
      gnomAD variants
    </h2>
    <p>
      No gnomAD variants found.
    </p>
  </div>,
]
`;

exports[`MitochondrialVariantsInTranscript with dataset gnomad_sv_r2_1_controls has no unexpected changes 1`] = `
[
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2>
      ClinVar variants
    </h2>
  </div>,
  <p
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    No ClinVar variants found in this transcript.
  </p>,
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2
      style={
        {
          "margin": "2em 0 0.25em",
        }
      }
    >
      gnomAD variants
    </h2>
    <p>
      No gnomAD variants found.
    </p>
  </div>,
]
`;

exports[`MitochondrialVariantsInTranscript with dataset gnomad_sv_r2_1_non_neuro has no unexpected changes 1`] = `
[
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2>
      ClinVar variants
    </h2>
  </div>,
  <p
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    No ClinVar variants found in this transcript.
  </p>,
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2
      style={
        {
          "margin": "2em 0 0.25em",
        }
      }
    >
      gnomAD variants
    </h2>
    <p>
      No gnomAD variants found.
    </p>
  </div>,
]
`;

exports[`MitochondrialVariantsInTranscript with dataset gnomad_sv_r4 has no unexpected changes 1`] = `
[
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2>
      ClinVar variants
    </h2>
  </div>,
  <p
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    No ClinVar variants found in this transcript.
  </p>,
  <div
    className="TrackPage__TrackPageSection-sc-1xq3qi7-1 fQVUsz"
  >
    <h2
      style={
        {
          "margin": "2em 0 0.25em",
        }
      }
    >
      gnomAD variants
    </h2>
    <p>
      No gnomAD variants found.
    </p>
  </div>,
]
`;
